src/performance/performance.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable, NgZone, ApplicationRef, InjectionToken, Inject, Optional } from '@angular/core';
+import { Observable } from 'rxjs';
+import { first, tap } from 'rxjs/operators';
+import { performance } from 'firebase/app';
+
+export const AUTOMATICALLY_TRACE_CORE_NG_METRICS = new InjectionToken<boolean>('angularfire2.performance.auto_trace');
+
+export type TraceOptions = {
+  metrics: {[key:string]: number},
+  attributes?:{[key:string]:string},
+  attribute$?:{[key:string]:Observable<string>},
+  incrementMetric$:{[key:string]: Observable<number|void|null|undefined>},
+  metric$?:{[key:string]: Observable<number>}
+};
+
+@Injectable()
+export class AngularFirePerformance {
+  
+  performance: performance.Performance;
+
+  constructor(
+    @Optional() @Inject(AUTOMATICALLY_TRACE_CORE_NG_METRICS) automaticallyTraceCoreNgMetrics:boolean|null,
+    appRef: ApplicationRef,
+    private zone: NgZone
+  ) {
+    
+    this.performance = zone.runOutsideAngular(() => performance());
```

Comment:
should server rendering be handled as a special case? could we add a custom attribute when we know this is being run on the server? otherwise server render times might skew the client-side results

Reply:
Good catch, I should test & likely gate around this. I may also need to create a seperate noop server class.

Reply:
I'm relatively confident this will blow up if included in a server build, just like messaging.

Reply:
I think we should punt on this for now & tell users to only include in their browser module. This is currently also a problem with messaging and we should sort out a build method to ensure that we don't break if they include either this or `AngularFireMessaging` in their server module.

I'll think about how we fix this and messaging in a build step... but I might wait for us to do the Bazel migration since we'll be fussing with everything then.